clients/src/main/java/org/apache/kafka/clients/ClientTelemetryUtils.java

```diff
@@ -0,0 +1,202 @@
+            }
+        }
+
+        log.debug("Error code: {}, reason: {}. Retry automatically in {} ms.", errorCode, reason, pushIntervalMs);
+        return Optional.of(pushIntervalMs);
+    }
+
+    public static Predicate<? super MetricKeyable> getSelectorFromRequestedMetrics(List<String> requestedMetrics) {
+        if (requestedMetrics == null || requestedMetrics.isEmpty()) {
+            log.debug("Telemetry subscription has specified no metric names; telemetry will record no metrics");
+            return SELECTOR_NO_METRICS;
+        } else if (requestedMetrics.size() == 1 && requestedMetrics.get(0) != null && requestedMetrics.get(0).isEmpty()) {
+            log.debug("Telemetry subscription has specified a single empty metric name; using all metrics");
+            return SELECTOR_ALL_METRICS;
+        } else {
+            log.debug("Telemetry subscription has specified to include only metrics that are prefixed with the following strings: {}", requestedMetrics);
+            return k -> requestedMetrics.stream().anyMatch(f -> k.key().name().startsWith(f));
+        }
+    }
+
+    public static List<CompressionType> getCompressionTypesFromAcceptedList(List<Byte> acceptedCompressionTypes) {
+        if (acceptedCompressionTypes == null || acceptedCompressionTypes.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        List<CompressionType> result = new ArrayList<>();
+        for (Byte compressionByte : acceptedCompressionTypes) {
+            int compressionId = compressionByte.intValue();
+            try {
+                CompressionType compressionType = CompressionType.forId(compressionId);
+                result.add(compressionType);
+            } catch (IllegalArgumentException e) {
+                log.warn("Accepted compressionByte type with ID {} is not a known compressionByte type; ignoring", compressionId, e);
+            }
+        }
+        return result;
+    }
+
+    public static Uuid validateClientInstanceId(Uuid clientInstanceId) {
+        if (clientInstanceId == null || clientInstanceId == Uuid.ZERO_UUID) {
+            throw new IllegalArgumentException("clientInstanceId is not valid");
+        }
+
+        return clientInstanceId;
+    }
+
+    public static int validateIntervalMs(int intervalMs) {
+        if (intervalMs <= 0) {
+            log.warn("Telemetry subscription push interval value from broker was invalid ({}),"
+                + " substituting with default value of {}", intervalMs, DEFAULT_PUSH_INTERVAL_MS);
+            return DEFAULT_PUSH_INTERVAL_MS;
+        }
+
+        log.debug("Telemetry subscription push interval value from broker: {}", intervalMs);
+        return intervalMs;
+    }
+
+    public static boolean validateRequiredResourceLabels(Map<String, String> metadata) {
+        return notEmptyString(metadata, MetricsContext.NAMESPACE);
+    }
+
+    public static boolean notEmptyString(Map<String, ?> m, String key) {
```

Comment:
Can we rename this for clarity? It seems to be checking the validity of the config given a key.

Reply:
Done.